token-metadata/js/src/instruction.ts

```diff
@@ -0,0 +1,205 @@
+    mint,
+    mintAuthority,
+    name,
+    symbol,
+    uri,
+}: Initialize): TransactionInstruction {
+    const values = { name, symbol, uri };
+    return new TransactionInstruction({
+        programId,
+        keys: [
+            { isSigner: false, isWritable: true, pubkey: metadata },
+            { isSigner: false, isWritable: false, pubkey: updateAuthority },
+            { isSigner: false, isWritable: false, pubkey: mint },
+            { isSigner: true, isWritable: false, pubkey: mintAuthority },
+        ],
+        data: Initialize.pack(values),
+    });
+}
+
+/** If the field does not exist on the account, it will be created.
+ * If the field does exist, it will be overwritten. */
+interface UpdateField {
+    programId: PublicKey;
+    metadata: PublicKey;
+    updateAuthority: PublicKey;
+    field: Field;
+    value: string;
+}
+
+export function createUpdateFieldInstruction({
+    programId,
+    metadata,
+    updateAuthority,
+    field,
+    value,
+}: UpdateField): TransactionInstruction {
+    const values = { field, value };
+    return new TransactionInstruction({
+        programId,
+        keys: [
+            { isSigner: false, isWritable: true, pubkey: metadata },
+            { isSigner: true, isWritable: false, pubkey: updateAuthority },
+        ],
+        data: UpdateField.pack(values),
+    });
+}
+
+/** Removes a key-value pair in a token-metadata account. This only applies
+ * to additional fields, and not the base name / symbol / URI fields. */
+export interface RemoveKey {
+    programId: PublicKey;
+    metadata: PublicKey;
+    updateAuthority: PublicKey;
+    field: Field;
+    idempotent: boolean;
+}
+
+export function createRemoveKeyInstruction({ programId, metadata, updateAuthority, field, idempotent }: RemoveKey) {
+    const values = { idempotent, field };
+    return new TransactionInstruction({
+        programId,
+        keys: [
+            { isSigner: false, isWritable: true, pubkey: metadata },
+            { isSigner: true, isWritable: false, pubkey: updateAuthority },
+        ],
+        data: RemoveKey.pack(values),
+    });
+}
+
+/** Updates the token-metadata authority */
+export interface UpdateAuthority {
+    programId: PublicKey;
+    metadata: PublicKey;
+    oldAuthority: PublicKey;
+    newAuthority: PublicKey;
+}
+
+export function createUpdateAuthorityInstruction({
+    programId,
+    metadata,
+    oldAuthority,
+    newAuthority,
+}: UpdateAuthority): TransactionInstruction {
+    const values = { newAuthority };
+    return new TransactionInstruction({
+        programId,
+        keys: [
+            { isSigner: false, isWritable: true, pubkey: metadata },
+            { isSigner: true, isWritable: false, pubkey: oldAuthority },
+        ],
+        data: UpdateAuthority.pack(values),
+    });
+}
+
+// Emits the token-metadata as return data
+export interface Emit {
+    programId: PublicKey;
+    metadata: PublicKey;
+    start?: bigint;
+    end?: bigint;
```

Comment:
Just want to highlight that i changed the type here to `bigint`. 

Realised `number` is smaller than `u64` it is being packed to

Reply:
Yeah perfect!